x/bank/keeper/keeper.go

```diff
@@ -53,12 +54,15 @@ type Keeper interface {
 type BaseKeeper struct {
 	BaseSendKeeper
 
-	ak         types.AccountKeeper
-	cdc        codec.BinaryCodec
-	storeKey   storetypes.StoreKey
-	paramSpace paramtypes.Subspace
+	ak                     types.AccountKeeper
+	cdc                    codec.BinaryCodec
+	storeKey               storetypes.StoreKey
+	paramSpace             paramtypes.Subspace
+	mintCoinsRestrictionFn BankMintingRestrictionFn
 }
 
+type BankMintingRestrictionFn func(ctx sdk.Context, coins sdk.Coins) error
```

Comment:
Can we put this in x/bank/types

Reply:
Sure! Was thinking the same thing! My only concern was where it should go though, didn't want to place it somewhere irrelevant, every in `types` seemed like they were there for a sole, clear purpose

Reply:
Oh thats a good point, seems fine to me here then